pkg/reconciler/route/resources/virtual_service.go

```diff
@@ -56,16 +86,22 @@ func MakeVirtualService(route *v1alpha1.Route) (*networking.VirtualService, erro
 		return nil, err
 	}
 
+	// Each route will own the VirtualService. Therefore none of them can be a
+	// controller.
+	ownerRef := *kmeta.NewControllerRef(route)
+	ownerRef.Controller = nil
+	ownerRef.BlockOwnerDeletion = nil
+
 	return &networking.VirtualService{
 		TypeMeta: metav1.TypeMeta{
 			APIVersion: "networking.istio.io/v1alpha3",
 			Kind:       "VirtualService",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      VirtualServiceName(route.Spec.Hostname, route.Spec.Domain, route.Spec.Path),
+			Name:      VirtualServiceName(route.Spec.Hostname, route.Spec.Domain),
 			Namespace: route.GetNamespace(),
 			OwnerReferences: []metav1.OwnerReference{
-				*kmeta.NewControllerRef(route),
```

Comment:
Since all routes own the service, should this be a merge operation? Maybe the MakeVirtualService method can take an optional, existing *networking.VirtualService.

Reply:
The merging happens in the reconciler.